### DISCUSSION TITLE
Problems with photon detection

### DISCUSSION THREAD

**Discussion by RecurvedBow** [fangq/mcx#226] [Help]
Hello everyone,

I am currently facing some troubles with photon detection: My goal is to define a planar illumination at z = 0 and detect the photons escaping from the top volume face. Therefore, all optical parameters of the volume were padded by 0 at the beginning of the z-Axis and I called MCX with the following flags for photon detection: --bc ______000001 --savedetflag XV --saveref. However, for some optical parameters, MCX does not detect any photons (0 photons detected) when using 10^7 photons and my program fails for not being able to find the output file *.jdat. For some reason, I never face this issue if I use -bc ______000010 instead, for which MCX does indeed detect returning photons. While I could keep using this alternate command, problems occur if I want to mirror the sides of the tissue on the x and y axis. mm_mm_000010 obviously does not work, while mm_mm_000001 has similar problems as ______000001. 

Is there something I have forgotten to check for photon detection or am I not using the flags correctly?

Thank you in advance for your answer.

**Comment by fangq** [comment]
@RecurvedBow, you don't have to pad zeros, or use `--saveref` if you want to capture detected photon using `--bc`, these are two different mechanisms to get diffuse transmittance - `--saveref` accumulates the dref for you, and `--bc` captures detected photons you post-process to get dref.

if the detected photon count via --bc is not what you have expected, please provide a sample code and describe the condition when the discrepancies happen

**Reply by RecurvedBow** [reply]
Thank you for your answer! I have investigated further and found out that "--savedetflag XV" was the actual problem as sometimes the photon position and direction on exit was not saved. If I can still replicate this issue even after your suggestion, I will provide a sample case then.

I have one more question: It seems like using only --bc without padding results in different reflectance/fluence values which I am used to. How do I accumulate the values such that I get the accumulated dref, like if using --saveref? 

Thank you again in advance!

**Reply by fangq** [reply]
> I have one more question: It seems like using only --bc without padding results in different reflectance/fluence values which I am used to. How do I accumulate the values such that I get the accumulated dref, like if using --saveref?

`--bc` only captures individual photon packets on the bounding box, you will have to use `mcxdetweight()` to get their exiting weights, and bin their exit positions to accumulate - but if you want an accumulated results, `--saveref` is the more efficient way. there is no need to use `--bc` in such case.

**Reply by RecurvedBow** [reply]
Thank you again for your answer! However, since I want the sides of the volume to reflect like a mirror, I will use `-bc mm_mm_000000 --saveref` for now. Is that the only option or is another intended way to capture the total photon count leaving a voxel while setting the behavior of the sides of the volumes to reflect the photons?

**Reply by fangq** [reply]
@RecurvedBow, it would be helpful if you can post a small code that I can reproduce the issue that "sometimes the photon position and direction on exit was not saved"

**Comment by fangq** [comment]
@
